### PR TITLE
java CDK: build no longer downloads files from connector registry

### PR DESCRIPTION
**Pull request by postamar**
These files were part of the CDK build and formed a circular dependency: their content is updated based on the connector metadata when a connector is published. Incidentally, removing these also allows removing the dependencies on micronaut.

This PR is best reviewed commit-by-commit:
- 95f6e26da5584db070e6e49a9cdb809dace68246 removes the whole init-oss module and changes `DestinationAcceptanceTest`, which is the only downstream dependency, to rely on the connector's `metadata.yaml` file instead of the connector metadata json blob.
- 40bf905b32413e7b44a362b792ef5da16c960c4d removes all references to micronaut in the repo; this is made possible by the previous commit.
- b25fe6fb02a2bc7b1a8c1782452ccafe42f040c3 removes the dependency on the `specs_secrets_mask.yaml` file which was used by a custom log4j plugin to scrub secrets from the logs. The log config in this repo is now only used for testing and since tests are run via airbyte-ci the secrets will already be scrubbed by dagger.
- the final two commits are version accounting boilerplate and also ensure the changes are validated on a couple of connectors.

### CODE REVIEW

**Review by stephane-airbyte — APPROVED** [review]

**Comment by postamar** [comment]
@alafanechere could you review my last commit 4e1a328 please?
This should ensure that the java logs never leak any secrets. I've tested this manually.

File: airbyte-ci/connectors/pipelines/pipelines/helpers/utils.py [hunk 1]
```diff
@@ -326,3 +327,28 @@ def fail_if_missing_docker_hub_creds(ctx: click.Context) -> None:
         raise click.UsageError(
             "You need to be logged to DockerHub registry to run this command. Please set DOCKER_HUB_USERNAME and DOCKER_HUB_PASSWORD environment variables."
         )
+
+
+def java_log_scrub_pattern(secrets_to_mask: List[str]) -> str:
```
Comment (alafanechere) [inline airbyte-ci/connectors/pipelines/pipelines/helpers/utils.py:332]: Could you please add a unit test for this? 🙏

**Review by alafanechere — COMMENTED** [review]
To what extent would it be harder / simpler to create this pattern within the containers under test instead of building the pattern in the Python logic?  

I'm also wondering if we could declare this logic within the `GradleTask` step.

Slightly out of scope:
Would it also be worth configuring the log scrubbing at the connector image level at runtime: when the config is parsed

File: airbyte-ci/connectors/pipelines/pipelines/helpers/utils.py (hunk 1 above)
Comment (postamar) [reply airbyte-ci/connectors/pipelines/pipelines/helpers/utils.py:332]: Yes! sorry it slipped my mind

**Comment by postamar** [comment]
Thanks for taking a look! I'll add the unit test.

> To what extent would it be harder / simpler to create this pattern within the containers under test instead of building the pattern in the Python logic?  

Unfortunately, the scrubbing needs to be applied to all logs, including the test framework, and not just the logs emitted by the connector container. Also, not all secrets are equal: the scrubbing should only apply to those secrets downloaded by ci_credentials, we don't want to scrub passwords for testcontainer instances (which furthermore are often dumb strings like "password" or "test").

> 
> I'm also wondering if we could declare this logic within the `GradleTask` step.

The integration test task class would be its proper place but for some reason, dagger secrets are defined by the PipelineContext, which isn't something I'm a fan of. This is why I moved the pattern generation logic to `util` instead of inlining it.

> 
> Slightly out of scope:
> Would it also be worth configuring the log scrubbing at the connector image level at runtime: when the config is parsed 

Indeed. @stephane-airbyte and I are also thinking about how to properly scrub secrets in prod. It's a bit more involved in that case.

**Comment by postamar** [comment]
I added a unit test and checked again that the scrubbing works (it does).

**Pull request merged**